### PR TITLE
Created new listener for adding new users to Mailchimp list

### PR DESCRIPTION
**Pull request by fillup**

### CODE REVIEW

File: application/marketing/marketing_test.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+					Email:     "integration-test@domainasdfasdf.com",
+					FirstName: "Integration",
+					LastName:  "Test",
+				},
+				apiBaseURL: domain.Env.MailChimpAPIBaseURL,
+				listId:     domain.Env.MailChimpListID,
+				username:   domain.Env.MailChimpUsername,
+				password:   domain.Env.MailChimpAPIKey,
+			},
+			wantErr: false,
```
Comment (Schparky) [inline application/marketing/marketing_test.go]: Did you want to add one or more tests where `wantErr` is `true`?

File: application/marketing/marketing_test.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+		apiRequests []ApiRequest
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "callApi tests",
+			wantErr: false,
```
Comment (Schparky) [inline application/marketing/marketing_test.go]: Did you want to add one or more tests where `wantErr` is true?

File: application/marketing/marketing_test.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+			args: args{
+				apiRequests: []ApiRequest{
```
Comment (Schparky) [inline application/marketing/marketing_test.go]: Not sure how you wanted this test to be scalable, but could you do only one request per test?

File: application/marketing/marketing.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	req.SetBasicAuth(apiRequest.Username, apiRequest.Password)
+	req.Header.Set("content-type", apiRequest.ContentType)
+	req.Header.Set("accept", "application/json")
+
+	for key, val := range apiRequest.Headers {
+		req.Header.Set(key, val)
+	}
+
+	client := &http.Client{
+		Timeout: ApiTimeout,
```
Comment (Schparky) [inline application/marketing/marketing.go]: I just copied this for the wecarry-lambdas code. Didn't realize the default timeout is not good.

File: application/marketing/marketing_test.go [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+		t.Run(tt.name, func(t *testing.T) {
+			for _, ar := range tt.args.apiRequests {
+				got, err := callApi(ar)
+				if (err != nil) != tt.wantErr {
+					t.Errorf("callApi() error = %v, wantErr %v", err, tt.wantErr)
+					return
+				}
+				var resp ApiRequest
+				err = json.Unmarshal([]byte(got), &resp)
+				if err != nil {
```
[the remark this answers is not in the document]
Comment (fillup) [reply application/marketing/marketing_test.go]: oh yeah

File: application/marketing/marketing_test.go [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+					t.Errorf("unable to unmarshal api response: %s", err.Error())
+					return
+				}
+				// ignore response headers for test
+				ar.Headers = map[string]string{}
+				resp.Headers = map[string]string{}
+				if !reflect.DeepEqual(resp, ar) {
```
[the remark this answers is not in the document]
Comment (fillup) [reply application/marketing/marketing_test.go]: yeah probably, I'll try that

File: application/marketing/marketing.go [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+		return "", err
+	}
+
+	bodyText, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("unable to read response body calling %s %s: %s",
+			apiRequest.Method, apiRequest.URL, err.Error())
+	}
+
+	if resp.StatusCode > 204 {
```
[the remark this answers is not in the document]
Comment (fillup) [reply application/marketing/marketing.go]: sure

File: application/marketing/marketing.go (hunk 4 above)
Comment (fillup) [reply application/marketing/marketing.go]: heh, even worse is to not realize the default "duration" is not seconds :-) I had the value at `3` for a while and spent 45 minutes trying to figure out why the connection was canceling constantly.

File: application/marketing/marketing.go [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+type MailChimpError struct {
+	Type     string `json:"type"`
+	Title    string `json:"title"`
+	Status   int    `json:"status"`
+	Detail   string `json:"detail"`
+	Instance string `json:"instance"`
+}
+
+func AddUserToList(u models.User, apiBaseURL, listId, username, password string) error {
```
[the remark this answers is not in the document]
Comment (fillup) [reply application/marketing/marketing.go]: i'm just going to leave it as is for now, it's clear and explicit and all the fields are required

File: application/marketing/marketing_test.go (hunk 2 above)
Comment (fillup) [reply application/marketing/marketing_test.go]: sure, I added one for that

File: application/marketing/marketing_test.go (hunk 3 above)
Comment (fillup) [reply application/marketing/marketing_test.go]: Thanks, I refactored it to not embed the multiple requests in a single test case

**Pull request merged**